examples/network_state_example.yml

```diff
@@ -0,0 +1,234 @@
+        bonding mode balance-rr
+      vars:
+        network_state:
+          interfaces:
+            - name: bond99
+              type: bond
+              state: up
+              link-aggregation:
+                mode: balance-rr
+                port:
+                  - eth1
+                  - eth2
+      ansible.builtin.include_role:
+        name: linux-system-roles.network
+
+    - name: Set the ports config for bond99, set the port eth1 priority to -1
+        and queue id to 1, set the port eth2 priority to 9 and queue id to 0
+      vars:
+        network_state:
+          interfaces:
+            - name: bond99
+              type: bond
+              state: up
+              link-aggregation:
+                ports-config:
+                - name: eth1
+                  priority: -1
+                  queue-id: 1
+                - name: eth2
+                  priority: 9
+                  queue-id: 0
+      ansible.builtin.include_role:
+        name: linux-system-roles.network
+
+
+    - name: Set the bonding mode balance-rr and miimon value to 200 for bond99
+      vars:
+        network_state:
+          interfaces:
+            - name: bond99
+              type: bond
+              state: up
+              link-aggregation:
+                mode: balance-rr
+                options:
+                  miimon: 200
+      ansible.builtin.include_role:
+        name: linux-system-roles.network
+
+    - name: Set the bonding mode to balance-tlb for bond device bond99
+      vars:
+        network_state:
+          interfaces:
+            - name: bond99
+              type: bond
+              state: up
+              link-aggregation:
+                mode: balance-tlb
+      ansible.builtin.include_role:
+        name: linux-system-roles.network
+
+    - name: Configure the ethernet device eth1 with dhcp4 configured
+      vars:
+        network_state:
+          interfaces:
+            - name: eth1
+              type: ethernet
+              state: up
+              ipv4:
+                enabled: true
+                dhcp: true
+      ansible.builtin.include_role:
+        name: linux-system-roles.network
+
+
+    - name: Set the default route to fe80::1 on eth1
+      vars:
+        network_state:
+          routes:
+            config:
+            - destination: ::/0
+              next-hop-interface: eth1
+              next-hop-address: fe80::1
+      ansible.builtin.include_role:
+        name: linux-system-roles.network
+
+    - name: Configure the ethernet device eth1 with dhcp4 configured
+      vars:
+        network_state:
+          interfaces:
+            - name: eth1
+              type: ethernet
+              state: up
+              ipv4:
+                enabled: true
+                dhcp: true
+      ansible.builtin.include_role:
+        name: linux-system-roles.network
+
+    - name: Configure the vrf interface test-vrf0 with vrf port eth1
```

Comment:
@eliaswimmer, this is the example of configuring vrf interface using the `network_state` variable.